---
title: "R Mini-Project Report"
author: "Mike Nsubuga"
date: "6/4/2021"
output:
  pdf_document: default
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## MSB7102 Mini-project, Semester 1, 2021

### Background
This is an assessed exercise that will contribute to coursework for this module. It is aimed at providing an experience with all that has been covered throughout the semester. The tasks are based on two Bioconductor packages; phyloseq and DEseq2. Endeavor to look at the documentation and the links indicated below may be useful. The data files are located in the shared google drive folder.

### Data sources and description
The data used in this exercise is derived and was generated by Kolistic et al in their study “The dynamics of the human infant gut microbiome in development and progression toward Type 1 Diabetes”. See full publication at https://doi.org/10.1016/j.chom.2015.01.001. Briefly, this was a prospective analysis of developing gut microbiome in infants en route to type 1 diabetes. Infants from Finland and Estonia were recruited at birth based on predisposition to autoimmunity determined by human leukocyte antigen (HLA) genotyping. The cohort consists of 33 infants, 11 of which seroconverted to serum autoantibody positivity and of those, four developed T1D within the three-year time-frame of this study.

### Tasks
1. Import the data described above into R, provide descriptive summaries of the subject data (using appropriate graphics and statistical summary measures) given in the diabimmune_16s_t1d_metadata.csv file. In addition, use appropriate test(s) to check for association/independency between disease status and other variables (delivery mode, gender and age). Note that age is given in days.
2. Using phyloseq, create a phyloseq object. This will comprise the OTU abundance, taxonomy (provided in the .txt file) and sample data (provided in the .csv file).
3. Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status.
4.	Perform differential abundance using DEseq2 
### Useful links

* Importing data: https://joey711.github.io/phyloseq/import-data.html 
* Ordination: https://joey711.github.io/phyloseq/plot_ordination-examples.html 
* Alpha diversity: https://joey711.github.io/phyloseq/plot_richness-examples.html 
* Differential abundance: http://joey711.github.io/phyloseq-extensions/DESeq2.html 



### Question 1
#### 1.1.1. Descriptive summaries of the subject data using appropriate graphics and statistical summary measures. <br>

Using the summary function to get a summary of descriptive statistics which include mean, median, 25th and 75th quartiles, min, max

```{r}
data <- read.csv("diabimmune_16s_t1d_metadata.csv");
summary(data)
```

```{r}
dim(data)
```

```{r}
class(data)

```

#### 1.1.2. Graphical summary representation of the data.     

Bar charts representing the number of occurences in each category

#### 1.1.3 Number of Cases and Controls against the Genders.

```{r}
library(ggplot2)
table(data$Case_Control)
table(data$Gender)

qplot(data$Case_Control, fill = data$Gender) + geom_bar() + labs(title = "A bar graph showing the Case_control against Gender", x = "Case_Control", y= "Number of people", fill = "Gender")

```

#### 1.1.4 The Delivery Route against the Gender
```{r}
library(ggplot2)
table(data$Case_Control)
table(data$Gender)

qplot(data$Delivery_Route, fill = data$Gender) + geom_bar() + labs(title = "A bar graph showing the Delivery routes against Gender", x = "Delivery Route", y= "Number of people", fill = "Gender")

```


#### 1.2.1. Appropriate tests to check for association/independency between disease status and other variables.


Two random variables x and y are called independent if the probability distribution of one variable is not affected by the presence of another.
 
Chi-square test examines whether rows and columns of a contigency table are statistically significatly associated

* Null hypothesis(H0): the row and the column variables of the contigency table are independent
* Alternate hypothesis(H1): row and column variables are dependent

##### Test the hypothesis whether the disease status is independent of their Delivery Mode at .05 significance level
```{r}
tbDelivery <- table(data$Case_Control, data$Delivery_Route)  #Generate a contigency table of Disease status and Delivery Mode
chisq.test(tbDelivery)

```
As the p-value 3.949e-09 is less than the .05 significance level, we reject the null hypothesis that the Disease Control status is independent of the Delivery Mode. This means that the two variables are somewhat related 

##### Test the hypothesis whether the disease status is independent of their Gender at .05 significance level.  

```{r}
tbGender <- table(data$Case_Control, data$Gender)  #Generate a contigency table of Disease status and Gender
chisq.test(tbGender)

```
As the p-value 0.5796 is greater than the .05 significance level, we do not reject the null hypothesis that the Disease Control status is independent of the Gender. It indicates strong evidence for the null hypothesis and reject the alternative hypothesis. This means that there's no relationship between the two variables being studied 

##### Test the hypothesis whether the disease status is independent of their Age_at_Collection at .05 significance level.

```{r}
tbAge <- table(data$Case_Control, data$Age_at_Collection)  #Generate a contigency table of Disease status and Age
chisq.test(tbAge)

```
As the p-value 0.6115 is greater than the .05 significance level, we do not reject the null hypothesis that the Disease Control status is independent of the Gender. It indicates strong evidence for the null hypothesis and reject the alternative hypothesis. This means that there's no relationship between the two variables being studied 

#### 1.2.2. Mosaic Plots.


To further visualise the tests of association/independecy, they can be plotted on a mosaic plot, which can give us insights on how the two variables are related.
Units are in standard deviations, so a residual greater than 2 or less than -2 represents a significant departure from independence. A mosaic plot maps standardised residuals to cells, positive sign is for blue and negative sign is for red ones. Blue means there are more observations in the cell than would be expected under the null model(independence). Red means there are fewer observations than would have been expected


##### A mosaic plot for Case Control vs Delivery Route
```{r}
library(graphics)
mosaicplot(~ Case_Control + Delivery_Route, data = data, main = "Case Control Vs Delivery Route", shade = TRUE)

```
From the plot, you will notice that the "control" cases are related to the method of delivery. Infact, most if not all of the contolled cases were delivered by cesarian method. 
Its slso important to note that there was a p-value less than .05 significance level when a chi-square test was conducted.
In other words, from the both tests, it is confident to say that there will be "controls" whenever the Delivery Route is cesarian.


##### A mosaic plot for Case Control vs Gender
```{r}
mosaicplot(~ Case_Control + Gender, data = data, main = "Case Control Vs Gender", shade = TRUE)

```
The plot indicates no significant relationship when you look at the standardised residuals


##### A mosaic plot for Case Control vs Age at Collection
```{r}
mosaicplot(~ Case_Control + Age_at_Collection, data = data, main = "Case Control Vs Age", shade = TRUE)

```

The plot indicates no significant relationship when you look at the standardised residuals

### Question 2
### Phyloseq


Using phyloseq, create a phyloseq object. this will comprise the OTU abundance, taxonomy(provided in the .txt file) and sample data(provided in the .csv file)

```{r}
otuTable <- read.table("otu_table")  #Importing the OTU table
#head(otuTable, n = 1)
```

```{r}
taxaTable <- read.table("taxa_table") #Importing the Taxa table
#head(taxaTable, n = 1)
```


```{r}
dim(otuTable)
```

```{r}
dim(taxaTable)
```

```{r}
class(otuTable)
```

```{r}
class(taxaTable)
```

```{r}
#Converting the Taxa and OTU Table into a Matrix
mtaxaTable <- as.matrix(taxaTable)
motuTable <- as.matrix(otuTable)

class(mtaxaTable)
class(motuTable)
```

```{r}
#Data Cleansing. This is done to have consistent data across all the matrices
#This will involve making sure that the OTU/taxa row names match. Currently they dont as taxa have a trailing ";"

#rownames(mtaxaTable)[rownames(mtaxaTable) == "4333897;"] = "4333897"

head(mtaxaTable, n=1)
tnames <- rownames(mtaxaTable)  #Extract rownames from the matrix
ntnames <- gsub(x = tnames, pattern = ";", replacement = "")  #Remove the ; from the extracted rownames
#ntnames
rownames(mtaxaTable) <- ntnames  #Set the new rownames
head(mtaxaTable, n=1)

```

```{r}
library(phyloseq)
library(ggplot2)

#Tell phyloseq to load them into a phyloseq object
OTU = otu_table(motuTable, taxa_are_rows = TRUE)
TAX = tax_table(mtaxaTable)

#OTU
#TAX

#Generating the phyloseq object
physeq = phyloseq(OTU, TAX)
physeq

#Plotting the phyloseq
plot_bar(physeq, fill = "Family.")
```


### Question 3. 
Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status

```{r}

plot_richness(physeq)  #Defauit plot produced by the plot_richness function

```

#### Alpha diversity comparison between the gender in cases and control
```{r}
#plot_richness(physeq = physeq, x ="Case_Control")

```
#### Alpha diversity comparison between the Delivery Mode and Case Control

```{r}
#plot_richness(physeq = physeq, x ="Case_Control")

```

### Question 4
Perform differential abundance using DESEq2
