---
title: "Mini project report"
author: "Mike Nsubuga"
date: "6/4/2021"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## MSB7102 Mini-project, Semester 1, 2021

### Background
This is an assessed exercise that will contribute to coursework for this module. It is aimed at providing an experience with all that has been covered throughout the semester. The tasks are based on two Bioconductor packages; phyloseq and DEseq2. Endeavor to look at the documentation and the links indicated below may be useful. The data files are located in the shared google drive folder.

### Data sources and description
The data used in this exercise is derived and was generated by Kolistic et al in their study “The dynamics of the human infant gut microbiome in development and progression toward Type 1 Diabetes”. See full publication at https://doi.org/10.1016/j.chom.2015.01.001. Briefly, this was a prospective analysis of developing gut microbiome in infants en route to type 1 diabetes. Infants from Finland and Estonia were recruited at birth based on predisposition to autoimmunity determined by human leukocyte antigen (HLA) genotyping. The cohort consists of 33 infants, 11 of which seroconverted to serum autoantibody positivity and of those, four developed T1D within the three-year time-frame of this study.

### Tasks
1. Import the data described above into R, provide descriptive summaries of the subject data (using appropriate graphics and statistical summary measures) given in the diabimmune_16s_t1d_metadata.csv file. In addition, use appropriate test(s) to check for association/independency between disease status and other variables (delivery mode, gender and age). Note that age is given in days.
2. Using phyloseq, create a phyloseq object. This will comprise the OTU abundance, taxonomy (provided in the .txt file) and sample data (provided in the .csv file).
3. Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status.
4.	Perform differential abundance using DEseq2 
### Useful links

* Importing data: https://joey711.github.io/phyloseq/import-data.html 
* Ordination: https://joey711.github.io/phyloseq/plot_ordination-examples.html 
* Alpha diversity: https://joey711.github.io/phyloseq/plot_richness-examples.html 
* Differential abundance: http://joey711.github.io/phyloseq-extensions/DESeq2.html 




### Descriptive summaries of the subject data using appropriate graphics and statistical summary measures
Using the summary function to get a summary of descriptive statistics which include mean, median, 25th and 75th quartiles, min, max

```{r}
data <- read.csv("diabimmune_16s_t1d_metadata.csv");
summary(data)
```

```{r}
dim(data)
```

```{r}
class(data)

```

### Graphical summary representation of the data
Bar charts representing the number of occurences in each category

#### Number of Cases and Controls
```{r}
library(ggplot2)
table(data$Case_Control)
qplot(data$Case_Control) + geom_bar()

```

#### Number of Males and Females
```{r}
table(data$Gender)
qplot(data$Gender) + geom_bar()

```
### Appropriate tests to check for association/independency between disease status and other variables 



### Phyloseq


Using phyloseq, create a phyloseq object. this will comprise the OTU abundance, taxonomy(provided in the .txt file) and sample data(provided in the .csv file)

```{r}
otuTable <- read.table("otu_table")  #Importing the OTU table
head(otuTable, n = 5)
```

```{r}
taxaTable <- read.table("taxa_table") #Importing the Taxa table
head(taxaTable, n = 5)
```


```{r}
dim(otuTable)
```

```{r}
dim(taxaTable)
```

```{r}
class(otuTable)
```

```{r}
class(taxaTable)
```

```{r}
#Converting the Taxa and OTU Table into a Matrix
mtaxaTable <- as.matrix(taxaTable)
motuTable <- as.matrix(otuTable)

class(mtaxaTable)
class(motuTable)
```

```{r}
#Data Cleansing. This is done to have consistet data across all the matrices
#This will involve making sure that the OTU/taxa row names match. Currently they dont as taxa have a trailing ";"

#rownames(mtaxaTable)[rownames(mtaxaTable) == "4333897;"] = "4333897"

head(mtaxaTable)
tnames <- rownames(mtaxaTable)  #Extract rownames from the matrix
ntnames <- gsub(x = tnames, pattern = ";", replacement = "")  #Remove the ; from the extracted rownames
#ntnames
rownames(mtaxaTable) <- ntnames  #Set the new rownames
head(mtaxaTable)

```

```{r}
library(phyloseq)
library(ggplot2)

#Tell phyloseq to load them into a phyloseq object
OTU = otu_table(motuTable, taxa_are_rows = TRUE)
TAX = tax_table(mtaxaTable)

#OTU
#TAX

#Generating the phyloseq object
physeq = phyloseq(OTU, TAX)
physeq

#Plotting the phyloseq
plot_bar(physeq, fill = "Family.")
```


## Question 3. 
Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status

```{r}

plot_richness(physeq)  #Defauit plot produced by the plot_richness function

```

### Alpha diversity comparison between the gender in cases and control
```{r}
#plot_richness(physeq = physeq, x ="Case_Control")

```


